custom_functions/find_related_containers.py

```diff
@@ -0,0 +1,128 @@
+        minimum_match_count = 1
+        
+    # If value list is equal to * then proceed to grab all indicator records for the current container
+    if isinstance(value_list, list) and value_list[0] == "*":
+        new_value_list = []
+        url = phantom.build_phantom_rest_url('container', current_container, 'artifacts') + '?page_size=0'
+        response_data = phantom.requests.get(uri=url, verify=False).json().get('data')
+        if response_data:
+            for data in response_data:
+                for k,v in data['cef'].items():
+                    if isinstance(v, list):
+                        for item in v:
+                            new_value_list.append(item)
+                    else:
+                        new_value_list.append(v)
+        indicator_id_list = fetch_indicator_ids(list(set(new_value_list)))
+    elif isinstance(value_list, list):
+        indicator_id_list = fetch_indicator_ids(value_list)
+    else:
+        raise TypeError(f"Invalid input for 'value_list': {value_list}")
+
+    # Get list of related containers
+    if indicator_id_list:
+        for indicator_id in list(set(indicator_id_list)):
+            url = phantom.build_phantom_rest_url() + 'indicator_common_container?indicator_ids={}'.format(indicator_id)
+            response_data = phantom.requests.get(url, verify=False).json()
+            
+            # Populate an indicator dictionary where the original ids are the dictionary keys and the                     
+            # associated continers are the values
+            if response_data:
+                indicator_id_dictionary[str(indicator_id)] = []
+                for item in response_data:
+                    indicator_id_dictionary[str(indicator_id)].append(item['container_id'])
+    else:
+        raise RuntimeError('Unable to fetch indicator_ids')
+        
+    # Iterate through the newly created indicator id dictionary and create a dictionary where 
+    # the keys are related containers and the values are the associated indicator ids
+    if indicator_id_dictionary:
+        for k,v in indicator_id_dictionary.items():
+            for item in v:
+                if str(item) not in container_dictionary.keys():
+                    container_dictionary[str(item)] = [str(k)]
+                else:
+                    container_dictionary[str(item)].append(str(k))
+    else:
+        raise RuntimeError('Unable to create indicator_id_dictionary')
+        
+    # Iterate through the newly created container dictionary                
+    if container_dictionary:
+        for k,v in container_dictionary.items():
+            # Dedupe the number of indicators
+            container_dictionary[str(k)] = list(set(v))
+            
+            # If any of the containers contain more than the minimum match count request that container detail.
+            if len(container_dictionary[str(k)]) > minimum_match_count:
```

Comment:
I think this should be greater than or equal to minimum_match_count. If a number barely meets the minimum threshold it is still big enough to count.

Reply:
Adjusting in next commit to be >=